src/main/java/backend/Logic.java

```diff
@@ -93,21 +95,54 @@ public void refresh() {
                 .exceptionally(Futures::log);
     }
 
+    /**
+     * Opens the repoId if it isn't already open, else simply refreshes the UI
+     * After opening the repo, it will trigger a PrimaryRepoOpenedEvent
+     *
+     * @param repoId
+     * @return
+     */
     public CompletableFuture<Boolean> openPrimaryRepository(String repoId) {
-        return openRepository(repoId, true);
+        return openRepository(repoId, true, Optional.empty());
     }
 
-    public CompletableFuture<Boolean> openRepositoryFromFilter(String repoId) {
-        return openRepository(repoId, false);
+    /**
+     * Opens the repoId if it isn't already open, else simply refreshes the UI
+     * After opening the repo, it will trigger a FilterOpeningEvent with panel as argument
+     *
+     * @param repoId
+     * @return
+     */
+    public CompletableFuture<Boolean> openRepositoryFromFilter(String repoId, FilterPanel panel) {
+        return openRepository(repoId, false, Optional.of(panel));
     }
 
-    private CompletableFuture<Boolean> openRepository(String repoId, boolean isPrimaryRepository) {
+    /**
+     * Opens the repoId if it isn't already open, else simply refreshes the UI
+     *
+     * After opening the repo, it will trigger a PrimaryRepoOpenedEvent or AppliedFilterEvent depending on the
+     * caller origin determined by isPrimaryRepository
+     *
```

Comment:
Explaining in terms `isPrimaryRepository` is sufficient I suppose. The caller method is kind of vague and doesn't really matter from this method's viewpoint.